python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -225,23 +315,48 @@ def _match_file(
     return index.query(s_type.hash_from_file(path))
 
 
+def _group_hashes_by_prefix(
+    path: pathlib.Path,
+    settings: CLISettings,
+    hashes_grouped_by_prefix: t.Dict[t.Optional[str], t.Set[str]],
+) -> None:
+    for line in path.read_text().splitlines():
+        line = line.strip()
+        if not line:
+            continue
+        components = line.split()
+        signal_type = None
+        if len(components) == 2:
+            # Assume it has a prefix
+            signal_type, hash = components
+        else:
+            # Assume it doesn't have a prefix and is a raw hash
+            hash = components[0]
+        hashes = hashes_grouped_by_prefix.get(signal_type, set())
+        hashes.add(hash)
+        hashes_grouped_by_prefix[signal_type] = hashes
```

Comment:
You can use setdefault again here to simplify (some folks also like defaultdict, but I have burned before!)

```
hashes_grouped_by_prefix.setdefault(signal_type, set()).add(hash)  # you are done
```
```